trax/tf_numpy/extensions/extensions.py

```diff
@@ -564,6 +565,93 @@ def tf_dot_general(lhs, rhs, dimension_numbers):
   return tf.einsum(equation, lhs, rhs)
 
 
+# TODO (Zhibo Zhang): Run pylint and complement the docstring.
+def _conv_general_param_type_converter(window_strides, lhs_dilation, rhs_dilation):
+  """ Convert the inputs strides, lhs_dilation, rhs_dilation to the standard
+  TF conv inputs.
+  For example,
+   in the 3D case, if lhs_dilation = 2, then convert it to [2, 2, 2]
+                   if lhs_dilation = (2, 2, 2), convert it also to [2, 2, 2]
+  """
+  strides = [window_strides] * dim if isinstance(window_strides, int) else \
```

Comment:
Where is dim coming from? 

Alternatively you can define an inner function and just use it on each:
```suggestion
def _as_list_of_size(item, size):
  if item is None:
    return None
  return [item] * size if isinstance(item, int) else list(item)
  
  return _as_list_of_size(window_strides), _as_list_of_size(lhs_dilation), _as_list_of_size(rhs_dilation)
```